types.go

```diff
@@ -1353,6 +1355,85 @@ func (rr *CSYNC) len(off int, compression map[string]struct{}) int {
 	return l
 }
 
+// APL RR. See RFC 3123.
+type APL struct {
+	Hdr      RR_Header
+	Prefixes []APLPrefix `dns:"apl"`
+}
+
+// APLPrefix is an address prefix hold by an APL record.
+type APLPrefix struct {
+	Negation bool
+	Network  net.IPNet
+}
+
+// String returns presentation form of the APL record.
+func (rr *APL) String() string {
+	sb := strings.Builder{}
+	sb.WriteString(rr.Hdr.String())
+	for i, p := range rr.Prefixes {
+		if i != 0 {
+			sb.WriteByte(' ')
+		}
+		sb.WriteString(p.String())
```

Comment:
Having a `String() string` method on both `APL` and `APLPrefix` both of which allocate isn't exactly great. Do we really need one on `APLPrefix`?

Reply:
Maybe not. I will make it private method. :+1: